### PR TITLE
[CARBONDATA-1146] Handling of the v3 format in delete flow of IUD.

### PR DESCRIPTION
**Pull request by ravikiran23**
problem : delete flow of IUD is not handling the V3 format feature of pages in the blocklet.

Solution: Adding the page info inside the delete delta file and handling of the same in the query flow.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/162/<h2>Build result: FAILURE</span></h2>[...truncated 80.67 KB...][JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/processing/pom.xml to org.apache.carbondata/carbondata-processing/1.2.0-SNAPSHOT/carbondata-processing-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Processing #161 to compare, so performing full copy of artifacts[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark-common/pom.xml to org.apache.carbondata/carbondata-spark-common/1.2.0-SNAPSHOT/carbondata-spark-common-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Spark Common #161 to compare, so performing full copy of artifacts[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/examples/spark/pom.xml to org.apache.carbondata/carbondata-examples-spark/1.2.0-SNAPSHOT/carbondata-examples-spark-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/hive/pom.xml to org.apache.carbondata/carbondata-hive/1.2.0-SNAPSHOT/carbondata-hive-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/common/pom.xml to org.apache.carbondata/carbondata-common/1.2.0-SNAPSHOT/carbondata-common-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Common #161 to compare, so performing full copy of artifacts[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/assembly/pom.xml to org.apache.carbondata/carbondata-assembly/1.2.0-SNAPSHOT/carbondata-assembly-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/pom.xml to org.apache.carbondata/carbondata-parent/1.2.0-SNAPSHOT/carbondata-parent-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Parent #161 to compare, so performing full copy of artifacts[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/examples/flink/pom.xml to org.apache.carbondata/carbondata-examples-flink/1.2.0-SNAPSHOT/carbondata-examples-flink-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/core/pom.xml to org.apache.carbondata/carbondata-core/1.2.0-SNAPSHOT/carbondata-core-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Core #161 to compare, so performing full copy of artifacts[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/hadoop/pom.xml to org.apache.carbondata/carbondata-hadoop/1.2.0-SNAPSHOT/carbondata-hadoop-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Hadoop #161 to compare, so performing full copy of artifacts[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/presto/pom.xml to org.apache.carbondata/carbondata-presto/1.2.0-SNAPSHOT/carbondata-presto-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark-common-test/pom.xml to org.apache.carbondata/carbondata-spark-common-test/1.2.0-SNAPSHOT/carbondata-spark-common-test-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark/pom.xml to org.apache.carbondata/carbondata-spark/1.2.0-SNAPSHOT/carbondata-spark-1.2.0-SNAPSHOT.pomNo artifacts from carbondata-pr-spark-1.6 » Apache CarbonData :: Spark #161 to compare, so performing full copy of artifactschannel stoppedSetting status of c641523280986c49b903db6d6c5f27615addd203 to FAILURE with url https://builds.apache.org/job/carbondata-pr-spark-1.6/162/ and message: 'Tests Failed for Spark1.6 'Using context: Jenkins(Spark1.6): mvn clean test -Pspark-1.6

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2289/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2293/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/166/<h2>Failed Tests: <span class='status-failure'>36</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark/testReport/org.apache.carbondata.spark.testsuite.datacompaction/DataCompactionTest/check_if_compaction_with_Updates/'><strong>org.apache.carbondata.spark.testsuite.datacompaction.DataCompactionTest.check if compaction with Updates</strong></a></li></ul><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>35</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from_carbon_table_with_alias__where_clause__/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from carbon table with alias [where clause ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_where_clause__/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[where clause ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_where_IN___/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[where IN  ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_with_alias_No_where_clause_/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[with alias No where clause]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_No_alias_No_where_clause_/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[No alias No where clause]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table__JOIN_with_another_table__/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[ JOIN with another table ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/DeleteCarbonTableTestCase/delete_data_from__carbon_table_where_numeric_condition___/'><strong>org.apache.carbondata.spark.testsuite.iud.DeleteCarbonTableTestCase.delete data from  carbon table[where numeric condition  ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_IUD_Horizontal_Compaction_Update_Alter_Clean/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test IUD Horizontal Compaction Update Alter Clean</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_IUD_Horizontal_Compaction_Delete/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test IUD Horizontal Compaction Delete</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_IUD_Horizontal_Compaction_Multiple_Update_Vertical_Compaction_and_Clean/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test IUD Horizontal Compaction Multiple Update Vertical Compaction and Clean</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_IUD_Horizontal_Compaction_Update_Delete_and_Clean/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test IUD Horizontal Compaction Update Delete and Clean</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_IUD_Horizontal_Compaction_Check_Column_Cardinality/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test IUD Horizontal Compaction Check Column Cardinality</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_IUD_Horizontal_Compaction_Segment_Delete_Test_Case/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test IUD Horizontal Compaction Segment Delete Test Case</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/HorizontalCompactionTestCase/test_case_full_table_delete/'><strong>org.apache.carbondata.spark.testsuite.iud.HorizontalCompactionTestCase.test case full table delete</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/test_update_operation_with_0_rows_updation_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.test update operation with 0 rows updation.</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_select_from_source_table_with_where_and_exist_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[select from source table with where and exist]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table_columns_with_where_and_exist_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table columns with where and exist]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_without_alias_in_set_columns/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table without alias in set columns</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_without_alias_in_set_columns_with_mulitple_loads/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table without alias in set columns with mulitple loads</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_without_alias_in_set_three_columns/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table without alias in set three columns</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_single_column_select_from_source_with_where_and_exist_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[single column select from source with where and exist]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_single_column_SELECT_from_source_with_where_and_exist_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[single column SELECT from source with where and exist]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table_columns_without_where_clause_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table columns without where clause]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table_columns_with_where_clause_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table columns with where clause]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table__no_alias__columns_without_where_clause_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table( no alias) columns without where clause]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table_columns_with_hard_coded_value__/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table columns with hard coded value ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_tableusing_destination_table_columns_with_hard_coded_value_and_where_condition_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon tableusing destination table columns with hard coded value and where condition]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_source__table_columns_with_where_and_exist_and_no_destination_table_condition_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using source  table columns with where and exist and no destination table condition]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table_where_and_exist_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table where and exist]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table__concat__where_and_exist_/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table (concat) where and exist]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon_table_using_destination_table__concat__with__where/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon table[using destination table (concat) with  where</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon__special_characters__in_value__test_parsing_logic__/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon [special characters  in value- test parsing logic ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon__sub_query__between_and_existing_in_outer_condition__Customer_query____/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon [sub query, between and existing in outer condition.(Customer query ) ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/update_carbon__self_join_select_query__/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.update carbon [self join select query ]</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/166/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.iud/UpdateCarbonTableTestCase/Failure_of_update_operation_due_to_bad_record_with_proper_error_message/'><strong>org.apache.carbondata.spark.testsuite.iud.UpdateCarbonTableTestCase.Failure of update operation due to bad record with proper error message</strong></a></li></ul>

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2295/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/168/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2299/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/172/

File: core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java [hunk 1]
```diff
@@ -80,8 +79,8 @@ private void initThreadPoolSize() {
    * @return
    * @throws Exception
    */
-  public int[] getDeleteDataFromAllFiles(List<String> deltaFiles, String blockletId)
-      throws Exception {
+  public Map<String, Integer[]> getDeleteDataFromAllFiles(List<String> deltaFiles,
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java]: Please maintain  `Map<Integer, int[]>`

File: core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java [hunk 2]
```diff
@@ -31,6 +31,8 @@
 
   private static final long serialVersionUID = 1206104914911491724L;
   private String id;
+  private String pageId;
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java]: Please maintain pageid as int, it is always integer

File: core/src/main/java/org/apache/carbondata/core/cache/update/BlockletLevelDeleteDeltaDataCache.java [hunk 3]
```diff
@@ -17,26 +17,36 @@
...
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+
 import org.roaringbitmap.RoaringBitmap;
 
 /**
  * This class maintains delete delta data cache of each blocklet along with the block timestamp
  */
 public class BlockletLevelDeleteDeltaDataCache {
-  private RoaringBitmap deleteDelataDataCache;
+  private Map<String, RoaringBitmap> deleteDelataDataCache =
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/cache/update/BlockletLevelDeleteDeltaDataCache.java]: change to `Map<Integer, RoaringBitmap>`

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2343/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/220/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/220/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/220/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.allqueries/InsertIntoCarbonTableTestCase/insert_into_carbon_table_from_carbon_table_union_query/'><strong>org.apache.carbondata.spark.testsuite.allqueries.InsertIntoCarbonTableTestCase.insert into carbon table from carbon table union query</strong></a></li></ul>

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/221/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/222/<h2>Build result: FAILURE</span></h2>[...truncated 78.91 KB...][ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :carbondata-spark[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/processing/pom.xml to org.apache.carbondata/carbondata-processing/1.2.0-SNAPSHOT/carbondata-processing-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark-common/pom.xml to org.apache.carbondata/carbondata-spark-common/1.2.0-SNAPSHOT/carbondata-spark-common-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/examples/spark/pom.xml to org.apache.carbondata/carbondata-examples-spark/1.2.0-SNAPSHOT/carbondata-examples-spark-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/hive/pom.xml to org.apache.carbondata/carbondata-hive/1.2.0-SNAPSHOT/carbondata-hive-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/common/pom.xml to org.apache.carbondata/carbondata-common/1.2.0-SNAPSHOT/carbondata-common-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/assembly/pom.xml to org.apache.carbondata/carbondata-assembly/1.2.0-SNAPSHOT/carbondata-assembly-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/pom.xml to org.apache.carbondata/carbondata-parent/1.2.0-SNAPSHOT/carbondata-parent-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/examples/flink/pom.xml to org.apache.carbondata/carbondata-examples-flink/1.2.0-SNAPSHOT/carbondata-examples-flink-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/core/pom.xml to org.apache.carbondata/carbondata-core/1.2.0-SNAPSHOT/carbondata-core-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/hadoop/pom.xml to org.apache.carbondata/carbondata-hadoop/1.2.0-SNAPSHOT/carbondata-hadoop-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/presto/pom.xml to org.apache.carbondata/carbondata-presto/1.2.0-SNAPSHOT/carbondata-presto-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark-common-test/pom.xml to org.apache.carbondata/carbondata-spark-common-test/1.2.0-SNAPSHOT/carbondata-spark-common-test-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/jenkins-slave/workspace/carbondata-pr-spark-1.6/integration/spark/pom.xml to org.apache.carbondata/carbondata-spark/1.2.0-SNAPSHOT/carbondata-spark-1.2.0-SNAPSHOT.pomchannel stoppedSetting status of 83097ce368cb35c42d20acba5e38b3a090267103 to FAILURE with url https://builds.apache.org/job/carbondata-pr-spark-1.6/222/ and message: 'Tests Failed for Spark1.6 'Using context: Jenkins(Spark1.6): mvn clean test -Pspark-1.6

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2344/

File: core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java (hunk 1 above)
Comment (ravikiran23) [inline core/src/main/java/org/apache/carbondata/core/reader/CarbonDeleteFilesDataReader.java]: DONE

File: core/src/main/java/org/apache/carbondata/core/cache/update/BlockletLevelDeleteDeltaDataCache.java (hunk 3 above)
Comment (ravikiran23) [inline core/src/main/java/org/apache/carbondata/core/cache/update/BlockletLevelDeleteDeltaDataCache.java]: DONE

File: core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java (hunk 2 above)
Comment (ravikiran23) [inline core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java]: DONE

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2345/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/223/

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/224/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**